### PR TITLE
[xmlsec] Update version to 1.3.5

### PR DESCRIPTION
**Pull request by metsma**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by dg0yt — COMMENTED** [review]
Originally I only wanted to mention a missing `find_dependency`. 
But the more I look at it, the more I wonder about other issues.
`find_dependency` is also missing for OpenSSL.

File: ports/xmlsec/CMakeLists.txt [hunk 1]
```diff
@@ -61,40 +60,32 @@ set(CMAKE_STATIC_LIBRARY_PREFIX "lib")
...
+include_directories(${CMAKE_CURRENT_BINARY_DIR}/include include)
 
 target_link_libraries(xmlsec1 PRIVATE
-    ${LIBXML2_LIBRARIES} OpenSSL::SSL
+    LibXml2::LibXml2 OpenSSL::Crypto
 )
 target_link_libraries(xmlsec1-openssl PRIVATE
-    ${LIBXML2_LIBRARIES} OpenSSL::SSL xmlsec1
+    LibXml2::LibXml2 OpenSSL::Crypto xmlsec1
 )
```
[the remark this answers is not in the document]
Comment (metsma) [reply ports/xmlsec/CMakeLists.txt]: Reverted this for now. Im not sure why the OpenSSL interface is not exported to config package.

File: ports/xmlsec/CMakeLists.txt [hunk 2]
```diff
@@ -61,40 +60,32 @@ set(CMAKE_STATIC_LIBRARY_PREFIX "lib")
 add_library(xmlsec1 ${SOURCESXMLSEC})
 add_library(xmlsec1-openssl ${SOURCESXMLSECOPENSSL})
 
-include_directories(${CMAKE_CURRENT_BINARY_DIR}/include include ${LIBXML2_INCLUDE_DIRS})
+include_directories(${CMAKE_CURRENT_BINARY_DIR}/include include)
 
 target_link_libraries(xmlsec1 PRIVATE
-    ${LIBXML2_LIBRARIES} OpenSSL::SSL
+    LibXml2::LibXml2 OpenSSL::Crypto
```
[the remark this answers is not in the document]
Comment (metsma) [reply ports/xmlsec/CMakeLists.txt]: Removed OpenSSL::Crypto dependency

File: ports/xmlsec/CMakeLists.txt [hunk 3]
```diff
@@ -146,14 +140,7 @@ if(INSTALL_HEADERS_TOOLS)
...
 	target_link_libraries(xmlsec PRIVATE
-		${LIBXML2_LIBRARIES} OpenSSL::SSL xmlsec1 xmlsec1-openssl
-	)
-    if(NOT Iconv_IS_BUILT_IN)
-        target_link_libraries(xmlsec PRIVATE Iconv::Iconv)
-    endif()
-    
-	target_compile_definitions(xmlsec PRIVATE
-		-DXMLSEC_CRYPTO_OPENSSL
+		LibXml2::LibXml2 OpenSSL::SSL xmlsec1 xmlsec1-openssl
```
[the remark this answers is not in the document]
Comment (metsma) [reply ports/xmlsec/CMakeLists.txt]: Probably all dependencies  are previously marked as PRIVATE

**Review by WangWeiLin-MV — APPROVED** [review]
The port installation tests pass with the following triplets:

* x64-windows

**Review by data-queue — APPROVED** [review]

**Pull request merged**